File: app/src/main/java/com/mahendri/pasbeli/entity/HargaKonsumen.java
package com.mahendri.pasbeli.entity;

import android.arch.persistence.room.ColumnInfo;
import android.arch.persistence.room.Entity;
import android.arch.persistence.room.PrimaryKey;

/**
 * @author Mahendri
 */

@Entity
@SuppressWarnings("unused")
public class HargaKonsumen {

    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "id_entry")
    public long idEntry;

    @ColumnInfo(name = "id_barang")
    public int idBarang;

    @ColumnInfo(name = "harga")
    public long harga;

    @ColumnInfo(name = "waktu_catat")
    public long waktuCatat;

    @ColumnInfo(name = "id_tempat")
    public String idTempat;

    @ColumnInfo(name = "nama_tempat")
    public String namaTempat;

    public int uploaded = 0;

    public double latitude;

    public double longitude;

    @Override
    public String toString() {
        return String.format("Nama barang %s, status upload %s", idBarang, uploaded);
    }
}

File: app/src/main/java/com/mahendri/pasbeli/entity/HargaKonsumen.kt
package com.mahendri.pasbeli.entity

import android.arch.persistence.room.ColumnInfo
import android.arch.persistence.room.Entity
import android.arch.persistence.room.PrimaryKey

/**
 * @author Mahendri
 */

@Entity
data class HargaKonsumen (
        @ColumnInfo(name = "id_barang")
        var idBarang: Int,
        @ColumnInfo(name = "harga")
        var harga: Long,
        @ColumnInfo(name = "waktu_catat")
        var waktuCatat: Long,
        @ColumnInfo(name = "nama_tempat")
        var namaTempat: String?,
        var mail: String?,
        var latitude: Double,
        var longitude: Double
) {
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "id_entry")
    var idEntry: Long = 0

    @ColumnInfo(name = "id_tempat")
    var idTempat: String? = null

    var uploaded: Int = 0
}